test/functional/ProgramCallFunctional.js

```diff
@@ -19,116 +19,92 @@
 /* eslint-disable new-cap */
 
 const { expect } = require('chai');
-const { readFileSync } = require('fs');
 const { parseString } = require('xml2js');
-const { ProgramCall } = require('../../lib/itoolkit');
-const { returnTransports } = require('../../lib/utils');
-
-// Set Env variables or set values here.
-let privateKey;
-if (process.env.TKPK) {
-  privateKey = readFileSync(process.env.TKPK, 'utf-8');
-}
-const opt = {
-  database: process.env.TKDB || '*LOCAL',
-  username: process.env.TKUSER || '',
-  password: process.env.TKPASS || '',
-  host: process.env.TKHOST || 'localhost',
-  port: process.env.TKPORT,
-  path: process.env.TKPATH || '/cgi-bin/xmlcgi.pgm',
-  privateKey,
-  passphrase: process.env.TKPHRASE,
-  verbose: !!process.env.TKVERBOSE,
-  dsn: process.env.TKDSN,
-};
-
-const transports = returnTransports(opt);
+const { ProgramCall, Connection } = require('../../lib/itoolkit');
+const { config } = require('./config');
+
 
 describe('ProgramCall Functional Tests', () => {
   describe('Test ProgramCall()', () => {
-    transports.forEach((transport) => {
-      it(`calls QWCRSVAL program checks if it ran successfully using ${transport.name} transport`, (done) => {
-        const connection = transport.me;
-
-        const program = new ProgramCall('QWCRSVAL', { lib: 'QSYS' });
-
-        const outBuf = [
-          [0, '10i0'],
-          [0, '10i0'],
-          ['', '36h'],
-          ['', '10A'],
-          ['', '1A'],
-          ['', '1A'],
-          [0, '10i0'],
-          [0, '10i0'],
-        ];
-
-        const errno = [
-          [0, '10i0'],
-          [0, '10i0', { setlen: 'rec2' }],
-          ['', '7A'],
-          ['', '1A'],
-        ];
-
-        program.addParam(outBuf, { io: 'out' });
-        program.addParam(66, '10i0');
-        program.addParam(1, '10i0');
-        program.addParam('QCCSID', '10A');
-        program.addParam(errno, { io: 'both', len: 'rec2' });
-        connection.add(program);
-        connection.run((error, xmlOut) => {
-          expect(error).to.equal(null);
-          parseString(xmlOut, (parseError, result) => {
-            expect(parseError).to.equal(null);
-            expect(result.myscript.pgm[0].success[0]).to.include('+++ success QSYS QWCRSVAL');
-            done();
-          });
+    it(`calls QWCRSVAL program checks if it ran successfully using ${config.transport} transport`, (done) => {
```

Comment:
I don't think we need to print out the transport used in the test description.

Might be nice to have in the test, though. Maybe dump the config (sans secrets) in the `before` callback? Could add a function to config that does this and just call it from `before` in each test suite.

Reply:
added `printConfig` function in `test/functional/config.js`.